redhat-developer/developers.redhat.com
# [MWES-2871] Replace Drupal database caching with memcached

This PR sets the ground-work for replacing Drupals default database caching with memcached. The following changes are introduced in this PR:

* Updated our image to use PHP 7.2 and installed the [memcached PECL](https://pecl.php.net/package/memcached)
* Installed the latest version of the memcached module in Drupal
* Extended the `redhat_developer` configuration object to include the `cache.engine` key. When this is set to `memcached`, then memcached is used as the cache engine of choice. By default this remains the database.
* Updated the local development experience to provide support for running memcached locally.

This PR does not yet enable memcached as the cache engine of choice. This is because we need to install the module first via `drush cim` and then switch the cache engine to be memcached. In addition I want to perform the migration to memcached in preview environments first to ensure it works as expected.

A follow up PR will enable memcached in preview environments and then prod.

### JIRA Issue Link
* https://issues.jboss.org/browse/MWES-2871

### Verification Process

* The build must go green
* Log in to Drupal and ensure that there is no dangling configuration
* Log in to Drupal and visit `/admin/reports/memcache` and ensure that it reports no memcached instances as in use
* Visit `/admin/reports/status` and ensure that it reports PHP as having `256M` of memory
* Log in to the container and verify that the following hosts are present in the `rhd.settings.php` as `memcached` cache nodes:
  * memcached-0.memcached-headless.rhdp--preview.svc.cluster.local
  * memcached-1.memcached-headless.rhdp--preview.svc.cluster.local
* [Log in to the container](https://github.com/redhat-developer/developers.redhat.com/tree/master/previews) and ensure that php version is `7.2.10`
  * run `php --version` from the command line
* [Log in to the container](https://github.com/redhat-developer/developers.redhat.com/tree/master/previews) and:
  * Edit `rhd.settings.php` and change the value of `$config['redhat_developer']['cache']['engine']` to `memcached`
  * `drush cr` from the command line
  * Revisit `/admin/reports/memcache` and notice now that there are two memcached nodes in-use
  * Browse the site and ensure things work as expected
* Visit the [Grafana Dashboard](https://grafana.util.phx2.redhat.com/d/XYrGhvDZk/developer-drupal?orgId=1), select Drupal Environment `preview` and ensure that metrics are being reported for both `memcached` instances
